integration_tests/steps/myco.py

```diff
@@ -27,6 +27,21 @@ def step_impl(context, setup_file: str, d3a_options: str):
            f"--no-export --seed 0 --enable-external-connection {d3a_options} ")
 
 
+@given('d3a-api-client is connected to simulation')
```

Comment:
This is a bit confusing to me. It is true that in this step the `d3a-api-client` connects to a simulation, but it's not so generic. You are connecting a specific aggregator (`AutoAggregator`, whose name should be changed IMO) which is different than all other aggregators provided by da3-api-client (it does some specific operations and checks).
In fact, if someone decides to replace this `AutoAggregator` with another aggregator class, [the other step you added below](https://github.com/gridsingularity/myco-api-client/pull/12/files#diff-30fb71d2e7defdcc73ef55468a34d585e54820a6f869a463c4aac72660e0ade9R75-R77) will raise an exception.

Reply:
I'm saying this because behave steps should be re-usable. If I change `context.aggregator` in a new step, I should be still able to call the other steps without failures (ideally).